### PR TITLE
Aroma

### PR DESCRIPTION
**Pull request by tiborauer**
This PR solves the issue with AROMA after its recent update.

### CODE REVIEW

**Comment by jones-michael-s** [comment]
Hi Tibor,

Thanks for working on this!

I saw the issue you opened on the AROMA repo in re 2.7 versus 3.6. Do that need to be resolved before this PR can be merged?

Also, I recall AROMA would not run here on anything but 2.7; other than the numpy dependency, has something changed in their code that fixes this?

**Comment by tiborauer** [comment]
No, we do not have to wait for that PR. My solution was to stick to an older commit, which does not ask for numpy=1.22.

You can try the installation I use for the CI. See the [updated aamod_AROMA_denoise.m](https://github.com/automaticanalysis/automaticanalysis/pull/353/files#diff-b46325ae1225fd2de95bbf193c611660a6e3e8f26a37b2cc744aca35bea4a4be) for instructions. Make sure you start from a new (conda) environment and you may consider switching off cache (i.e. `pip install --no-cache-dir -r requirements.txt`) to simulate a 'virgin' environment.

**Comment by jones-michael-s** [comment]
Downloaded this PR to test locally. Unfortunately, it doesn't get past aas_processBIDS:

```
Error using matlabpath
Must be a string scalar or character vector.

Error in rmpath (line 79)
	matlabpath([cmdirs{:}])

Error in toolboxClass/unload (line 147)
                rmpath(sprintf(['%s' pathsep],obj.toolInPath{:}))

Error in aas_processBIDS (line 159)
SPMtool.unload;
```

Stepping through rmpath in the debugger, it appears SPMtool.unload has deleted my entire Matlab path (therefore cmdirs{:} is empty at rmpath line 79, which matlabpath doesn't like).

Is there any chance you may have run across this error in the past and remember the fix? It's probably not an aas_processBIDS.m or toolboxClass.m issue because the versions of these files in this PR are identical to those in the older aa version I'm using here for production work which doesn't crash.

M

**Comment by jones-michael-s** [comment]
A bit more info: The problem seems to be related to spmClass.m. Currently load reads

```
        function load(obj)
            addpath(obj.toolPath);
            
            load@toolboxClass(obj)
            
            spm_jobman('initcfg');
            
            % Update stored path
            p = split(path,pathsep);                
            obj.toolInPath = cat(1,...
                obj.toolInPath,...
                setdiff(p(cellfun(@(x) contains(x,obj.toolPath), p)),obj.toolInPath)...
                );
```
 
Here, obj.toolInPath  =  '/path/to/automaticanalysis/aa_tools/toolboxes/spm_mods' on entry, but it gets set to my entire matlab path contents (1085 directories) by the update (obj.toolInPath = cat(1,...). The SPMtool.unload that gets called when aas_processBIDS finishes tries to remove all of these, which leads to the crash.

FYI: The previous version of spmClass.m (which does not crash for me) does not update toolInPath:

```
        function load(obj)
            addpath(obj.toolPath);
            spm_jobman('initcfg');
            
            load@toolboxClass(obj)
        end
```


(Sorry for being derpy about this -- I'm just not familiar with the toolbox class code.)

**Comment by jones-michael-s** [comment]
UPDATE: Ok, I found the problem (it's was an incompatibility in my param file)

Will begin testing the AROMA changes...

**Comment by jones-michael-s** [comment]
Ok, I was able to get this to run on my machine using our denoising example script and my existing install of the AROMA toolbox (should be the same as git reset --hard e8d7a5891563aff7ff71375078314c417d32d808).  I actually tested it under both Python 2.7 and 3.6. Both work.

TLDR: PR looks good to go. Merge away.

**Comment by tiborauer** [comment]
Thank you, Mike, for the meticulous testing.
